src/join-8-auth.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// join-8-auth.sol -- Non-standard token adapters
+
+// Copyright (C) 2018 Rain <rainbreak@riseup.net>
+// Copyright (C) 2018-2020 Maker Ecosystem Growth Holdings, INC.
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.6.12;
+
+interface VatLike {
+    function slip(bytes32, address, int256) external;
+}
+
+interface GemLike {
+    function decimals() external view returns (uint8);
+    function transfer(address,uint256) external returns (bool);
+    function transferFrom(address,address,uint256) external returns (bool);
+    function erc20Impl() external view returns (address);
+}
+
+// AuthGemJoin8
+// For a token that has a lower precision than 18, has decimals and it is upgradable (like GUSD)
+
+contract AuthGemJoin8 {
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
```

Comment:
nit: rely/deny could be formatted in one line to match join-5-auth

Reply:
I think we opted to instead convert to the other way.

Reply:
we should format the `join-5-auth` accordingly then, to keep them consistent.

Reply:
Done in #4